src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/StartupCode/StartupCodeHelpers.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.CompilerServices;
+using System.Security;
+using Internal.NativeFormat;
+using System.Diagnostics.Contracts;
+using System.Runtime.InteropServices;
+using System.Text;
+using System;
+using System.Runtime;
+
+namespace Internal.Runtime.CompilerHelpers
+{
+    internal static class StartupCodeHelpers
+    {
+        internal static string[] CommandLineArgs { get; set; }
+
+        internal static void Initialize()
+        {
+            InitializeStringTable();
+        }
+
+        internal static void Shutdown()
+        {
+        }
+
+        internal static unsafe void InitializeCommandLineArgsW(int argc, IntPtr argv)
+        {
+            string[] args = new string[argc];
+            for (int i = 0; i < argc; ++i)
+            {
+                char* argval = ((char**)argv)[i];
```

Comment:
You can just make the argument to be char** - avoids the cast here, and makes it clear what the method takes as input.